lti_consumer/api.py

```diff
@@ -34,15 +38,26 @@ def _get_or_create_local_lti_config(lti_version, block_location, external_id=Non
     # The create operation is only performed when there is no existing configuration for the block
     lti_config, _ = LtiConfiguration.objects.get_or_create(location=block_location)
 
-    if lti_config.version != lti_version:
-        lti_config.version = lti_version
-
-    if external_id:
-        lti_config.config_store = LtiConfiguration.CONFIG_EXTERNAL
-        lti_config.external_id = external_id
+    config_store = None
+    if config_type == "external" or external_id is not None:
+        config_store = LtiConfiguration.CONFIG_EXTERNAL
+    elif config_type == "database":
+        config_store = LtiConfiguration.CONFIG_ON_DB
     else:
-        lti_config.config_store = LtiConfiguration.CONFIG_ON_XBLOCK
-        lti_config.external_id = None
+        config_store = LtiConfiguration.CONFIG_ON_XBLOCK
+
+    lti_config.config_store = config_store
+    lti_config.external_id = external_id
+
+    # If the LtiConfiguration version on the model does not match what is on the configuration used by the integration
+    # (i.e xBlock or external configuration), then update the model to match. This may occur if the LTI
+    # configuration is stored on the xBlock (i.e. "new"), and the xBlock field has changed since the LtiConfiguration
+    # was last accessed.
+    # If the LTI configuration is stored on the LtiConfiguration model (i.e. "database"), then the lti_version function
+    # parameter can be ignored. There's no need to update the version on the model, because the model is the source of
```

Comment:
what does it mean if the database says version 1.3 but the incoming change says version 1.1? Doesn't that mean the request is possibly incoherent?

Reply:
does this actually get solved by the calls to `get_or_update_config` in #254? This kinda feels like an unexpected side-effect to "get or create" but I'm not sure where it belongs.